File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dic = {}
    for cloth in clothes:
        a, b = cloth
        if b not in dic:
            dic[b] = 1
        else:
            dic[b] += 1

    answer = 1
    for value in dic.values():
        answer *= (value + 1)

    return answer - 1 

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 0;
    map<string, int> dic;
    for(int i=0;i<clothes.size();i++){
        string name = clothes[i][0];
        string idx = clothes[i][1];
        dic[idx]++;
    }
    int count = 1;
    for(auto it : dic){
        count *= (it.second+1);
    }
    answer = count-1;
    
    return answer;
}